### PR TITLE
Update to JLab 3

### PR DESCRIPTION
**Pull request by ian-r-rose**
Supersedes #187.

### CODE REVIEW

File: binder/start [hunk 1]
```diff
@@ -1,9 +1,13 @@
 #!/bin/bash
 
 # Replace DASK_DASHBOARD_URL with the proxy location
-sed -i -e "s|DASK_DASHBOARD_URL|${JUPYTERHUB_BASE_URL}user/${JUPYTERHUB_USER}/proxy/8787|g" binder/jupyterlab-workspace.json
+sed -i -e "s|DASK_DASHBOARD_URL|${JUPYTERHUB_BASE_URL}user/${JUPYTERHUB_USER}/proxy/8787|g" binder/example.jupyterlab-workspace
 
+# TODO: Importing of workspace via CLI is currently broken.
+# remove this workaround once https://github.com/jupyterlab/jupyterlab/issues/10805 is resolved
+mkdir -p ~/.jupyter/lab/workspaces/
+mv binder/example.jupyterlab-workspace ~/.jupyter/lab/workspaces/default-37a8.jupyterlab-workspace
```
Comment (jrbourbeau) [inline binder/start:9]: The `37a8` part of this path seems quite specific. Why is that part needed? 

```suggestion
mv binder/example.jupyterlab-workspace ~/.jupyter/lab/workspaces/default-37a8.jupyterlab-workspace
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @ian-r-rose! I've left a couple of questions but will defer to you on all things JupyterLab : )

The new `binder/example.jupyterlab-workspace` workspace file looks very similar (though differently formatted) to the old `binder/jupyterlab-workspace.json` file. What's the meaningful difference between the two?

**Comment by jrbourbeau** [comment]
FYI there are unrelated CI failures on `main` (xref https://github.com/dask/dask-examples/pull/215), so I don't anticipate CI will pass on this PR

File: binder/start (hunk 1 above)
Comment (ian-r-rose) [reply binder/start:9]: This is 100% a hack -- that's a hash based on the workspace name which is not guaranteed to be stable. But I think it might be okay for the time-being, and is better than the currently broken state.

Comment (jrbourbeau) [reply binder/start:9]: Gotcha -- totally fine with me if it works

**Comment by ian-r-rose** [comment]
> What's the meaningful difference between the two?

There might not be a meaningful difference between them. Certainly one of the problems that I ran into is that some of the dashboard panels have changed names, but it depends on the specific layout.

**Comment by ian-r-rose** [comment]
> some of the dashboard panels have changed names

Which reminds me: we should check this again once we update `distributed`

**Comment by bryanwweber** [comment]
@jrbourbeau Yes, I'm working on completing #215

File: binder/environment.yml [hunk 2]
```diff
@@ -7,9 +7,8 @@ dependencies:
   - dask=2.20.0
   - dask-image=0.2.0
   - dask-ml=1.6.0
-  - dask-labextension=2.0.2
-  - jupyterlab=2.1
-  - nodejs=16
+  - dask-labextension>=5.2.0
+  - jupyterlab>=3
```
Comment (bryanwweber) [inline binder/environment.yml:11]: This introduces some conflicts with #218, so this should be merged first then I can fix that one.

**Review by jrbourbeau — APPROVED** [review]
Thanks @ian-r-rose -- let's give this a try!
